snippets.py

```diff
@@ -725,80 +742,78 @@ def _maybe_send_snippets_mail(to, subject, template_path, template_values):
+        if not has_snippet:
+            _send_mail(user_email)
+            logging.debug('sent reminder email to %s', user_email)
+        else:
+            logging.debug('did not send reminder email to %s: '
+                          'has a snippet already', user_email)
 
-        msg = 'Reminder: Weekly snippets due today at 5pm.'
-        _send_to_chat(msg, "/")
+    msg = 'Reminder: Weekly snippets due today at 5pm.'
+    _send_to_chat(msg, "/")
+    return flask.make_response({"status": 200, "message": "Sent reminder emails"})
 
 
-class SendViewEmail(BaseHandler):
+@app.route("/admin/send_view_email")
+def admin_send_view_email_handler() -> flask.Response:
     """Send an email to everyone to look at the week's snippets."""
 
-    def _send_mail(self, email, has_snippets):
+    def _send_mail(email, has_snippets):
         template_values = {'has_snippets': has_snippets}
         _maybe_send_snippets_mail(email, 'Weekly snippets are ready!',
                                   'view_email.txt', template_values)
 
-    def get(self):
-        email_to_has_snippet = _get_email_to_current_snippet_map(_TODAY_FN())
-        for (user_email, has_snippet) in email_to_has_snippet.iteritems():
-            self._send_mail(user_email, has_snippet)
-            logging.debug('sent "view" email to %s' % user_email)
-
-        msg = 'Weekly snippets are ready!'
-        _send_to_chat(msg, "/weekly")
-
-
-application = webapp2.WSGIApplication([
-    ('/', UserPage),
-    ('/weekly', SummaryPage),
-    ('/update_snippet', UpdateSnippet),
-    ('/settings', Settings),
-    ('/update_settings', UpdateSettings),
-    ('/admin/settings', AppSettings),
-    ('/admin/update_settings', UpdateAppSettings),
-    ('/admin/manage_users', ManageUsers),
-    ('/admin/send_friday_reminder_chat', SendFridayReminderChat),
-    ('/admin/send_reminder_email', SendReminderEmail),
-    ('/admin/send_view_email', SendViewEmail),
-    ('/admin/test_send_to_hipchat', hipchatlib.TestSendToHipchat),
-    ('/slack', slacklib.SlashCommand),
-    ],
-    debug=True)
+    email_to_has_snippet = _get_email_to_current_snippet_map(_TODAY_FN())
+    for user_email, has_snippet in email_to_has_snippet.items():
+        _send_mail(user_email, has_snippet)
+        logging.debug('sent "view" email to %s', user_email)
+
+    msg = 'Weekly snippets are ready!'
+    _send_to_chat(msg, "/weekly")
+    return flask.make_response({"status": 200, "message": "Sent 'snippets are ready' emails"})
+
+
+@app.route('/_ah/warmup')
+def warmup():
+    """App engine warmup requests handler
+
+    See https://cloud.google.com/appengine/docs/standard/configuring-warmup-requests?tab=python
+    """
+    return "OK", 200, {}
+
+
+app.add_url_rule("/slack", view_func=slacklib.slash_command_handler,
+                 methods=["POST"])
+app.add_url_rule("/admin/test_send_to_hipchat",
+                 view_func=hipchatlib.test_send_to_hipchat_handler)
```

Comment:
This is a bit awkward.  Would it make more sense to put the app in a separate app.py file, and let slacklib and hipchatlib import app.py and use the normal `@app.route`?

Reply:
I wouldn't spend a huge amount of time on this, just something to think about.

Reply:
That turned out to be nice and simple; done.